src/lib/client/Client.ts

```diff
@@ -347,3 +360,72 @@ export abstract class Plugin {
+	[ClientEvents.GuildDelete](guild: Guild): void;
+	[ClientEvents.GuildEmojiCreate](emoji: GuildEmoji, guild: Guild): void;
+	[ClientEvents.GuildEmojiDelete](emoji: GuildEmoji, guild: Guild): void;
+	[ClientEvents.GuildEmojiUpdate](emoji: GuildEmoji, previousEmoji: GuildEmoji, guild: Guild): void;
+	[ClientEvents.GuildIntegrationsUpdate](guild: Guild, previousGuild: Guild): void;
+	[ClientEvents.GuildMemberAdd](guildMember: GuildMember): void;
+	[ClientEvents.GuildMemberRemove](guildMember: GuildMember): void;
+	[ClientEvents.GuildMembersChunk](members: GuildMember[], guild: Guild, extraData: { chunkCount?: number, chunkIndex?: number, nonce?: number }): void;
+	[ClientEvents.GuildMemberUpdate](guildMember: GuildMember, previousGuildMember: GuildMember): void;
+	[ClientEvents.GuildRoleCreate](role: Role): void;
+	[ClientEvents.GuildRoleDelete](role: Role): void;
+	[ClientEvents.GuildRoleUpdate](role: Role, previousRole?: Role): void;
+	[ClientEvents.GuildUpdate](guild: Guild, previousGuild?: Guild): void;
+	[ClientEvents.InviteCreate](invite: Invite): void;
+	[ClientEvents.InviteDelete](invite: Invite): void;
+	[ClientEvents.MessageCreate](message: Message): void;
+	[ClientEvents.MessageDelete](message: Message, previousMessage?: Message): void;
+	[ClientEvents.MessageDeleteBulk](messages: ({ id: string } | Message)[], channel: TextBasedChannel): void;
+	[ClientEvents.MessageReactionAdd](reaction: MessageReaction, user: User): void;
+	[ClientEvents.MessageReactionRemove](reaction: MessageReaction, user: User): void;
+	[ClientEvents.MessageReactionRemoveAll](message: Message, reactions: Cache<string, MessageReaction>): void;
+	[ClientEvents.MessageReactionRemoveEmoji](reaction: MessageReaction): void;
+	[ClientEvents.MessageUpdate](message: Message, previousMessage: Message): void;
+	[ClientEvents.PieceDisabled](piece: Piece): void;
+	[ClientEvents.PieceEnabled](piece: Piece): void;
+	[ClientEvents.PieceLoaded](piece: Piece): void;
+	[ClientEvents.PieceReloaded](piece: Piece): void;
+	[ClientEvents.PieceUnloaded](piece: Piece): void;
+	[ClientEvents.PresenceUpdate](presence: Presence, previousPresence: Presence | null): void;
+	[ClientEvents.Ratelimited](...args: unknown[]): void;
+	[ClientEvents.Ready](): void;
+	[ClientEvents.RESTDebug](message: string): void;
+	[ClientEvents.Resumed](): void;
+	[ClientEvents.ShardOnline](): void;
+	[ClientEvents.ShardReady](shard: WebSocketShard): void;
+	[ClientEvents.ShardResumed](shard: WebSocketShard): void;
+	[ClientEvents.TypingStart](channel: GuildBasedChannel | DMChannel, user: User): void;
+	[ClientEvents.UserUpdate](user: User): void;
+	[ClientEvents.VoiceServerUpdate](voiceServerData: VoiceServerUpdateDispatch): void;
+	[ClientEvents.VoiceStateUpdate](voiceState: VoiceState): void;
+	[ClientEvents.WebhooksUpdate](channel: TextChannel | NewsChannel, previousChannel: TextChannel | NewsChannel): void;
+	[ClientEvents.WSDebug](message: string): void;
+	[ClientEvents.WTF](message: string): void;
+}
```

Comment:
This should be changed to the respective names, see https://github.com/microsoft/TypeScript/issues/38759#issuecomment-638590133

We target TypeScript 3.9.x (stable) as it's what people use, and using nightly builds will cause issues for anyone not using it. For now, using the names fix the untyped problem when using strings instead of the enum.

Reply:
![image](https://user-images.githubusercontent.com/4019718/83972637-94749e00-a8e1-11ea-88ee-d6f842a04384.png)
Confirmed working after 034c124a1a15235ea96ff679d9e037fccb90d0fa